### ISSUE TITLE
Debatable Style/RedundantArgument recommended value for Array#sum

### ISSUE THREAD

**Issue by gorodulin** [rubocop/rubocop#11531]
## Expected behavior

`Style/RedundantArgument` should not raise warnings for `array.sum(0)` by default

## Actual behavior

The cop returns by default:
`Style/RedundantArgument: Argument 0 is redundant because it is implied by default.`

Seems `Style/RedundantArgument` recommended approach for `Array#sum` method is incorrect.

Documentation [states](https://msp-greg.github.io/rubocop/RuboCop/Cop/Style/RedundantArgument.html):
```
# bad
array.sum(0)
# good
array.sum
```

However,
It is safer when the default value is defined explicitly otherwise it may affect the type of value returned.

Example:
```ruby
# Example 1
expenses.sum
=> 0.0 # Float when expenses == [-3.0,2.0,1.0]

# Example 2
expenses.sum
=> 0 # Integer when expenses array is empty
```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
% rubocop -V
1.43.0 (using Parser 3.2.0.0, rubocop-ast 1.24.1, running on ruby 3.0.3) [x86_64-darwin21]
  - rubocop-performance 1.15.1
  - rubocop-rails 2.17.3
  - rubocop-rspec 2.16.0
  - rubocop-thread_safety 0.4.4
```

**Comment by dvandersluis** [comment]
I'm not sure I follow your examples.

```ruby
[-3.0,2.0,1.0].sum
=> 0.0

[-3.0,2.0,1.0].sum(0)
=> 0.0

[-3.0,2.0,1.0].sum(0.0)
=> 0.0
```

Can you give an example where the default `0` changes the return?

**Comment by vlad-pisanov** [comment]
@dvandersluis he's referring to the empty list case
```ruby
[].sum.class    # Integer
[0.1].sum.class # Float
```

**Comment by koic** [comment]
I'm not sure about case where this is unsafe. The default argument value for `Array#sum` is 0:

> sum(init = 0) → objectclick to toggle source
> sum(init = 0) {|element| ... } → object

https://ruby-doc.org/core-3.1.0/Array.html#method-i-sum

For the following case:

> ```ruby
> [].sum.class    # Integer (0)
> [0.1].sum.class # Float (0.1)
> ```

Equivalent even with an argument of 0:

```ruby
[].sum(0).class    #=> Integer (0)
[0.1].sum(0).class #=> Float (0.1)
```

**Comment by vlad-pisanov** [comment]
The cop itself is safe, I think what @gorodulin is saying that `sum` should be omitted from it by default due to _potential_ type confusion/misuse by the end user. 

This seems more like a personal preference. `sum` can be excluded via the `Methods` configuration attribute.

**Comment by koic** [comment]
I'm going to close this issue because `Array#sum`'s default value can be omitted according to the API doc. Please reopen if you have any issues for that.

**Comment by FunnyHector** [comment]
I'd like to point out one case where we might not want to flag `sum(0)`.

Rails 7 started to deprecate `Enumberable#sum` without initial value when summing non-numeric elements.

Example:

```
require 'money'

> [Money.from_cents(5, "USD"), Money.from_cents(50, "USD")].sum
ActiveSupport::DeprecationException: DEPRECATION WARNING: Rails 7.0 has deprecated Enumerable.sum in favor of Ruby's native implementation available since 2.4. Sum of non-numeric elements requires an initial argument. (called from block (3 levels) in <top (required)> at ...)
from /.../.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/activesupport-7.0.4.2/lib/active_support/core_ext/enumerable.rb:315:in `sum'

> [Money.from_cents(5, "USD"), Money.from_cents(50, "USD")].sum(0)
{
           :cents => 55,
    :currency_iso => "NZD"
}
```
(I use [`Money` objects](https://github.com/RubyMoney/money) in this example as this is a real example in my codebase, but you could reproduce the same behaviour on any object that has implemented coercion and can add to 0).

For now I can adjust `Methods` config to ignore `sum`, but feels like this cop is going to conflict Rails 7.

**Comment by koic** [comment]
Basically, RuboCop is targeting vanilla Ruby. This will be a feedback to RuboCop Rails or Rails.

**Comment by koic** [comment]
@FunnyHector That deprecation warning to collections of `Money` objects will be resolved in the next Rails 7.0.x release:
https://github.com/rails/rails/commit/7561403bc040487d9ebb50811163157e938407c2

**Comment by FunnyHector** [comment]
Cool, awesome, thanks! I think that fix would resolve this perfectly.

**Comment by itsNikolay** [comment]
I'm getting following warning if I use `.sum`
```
DEPRECATION WARNING: Rails 7.0 has deprecated Enumerable.sum in favor of Ruby's native implementation available since 2.4. Sum of non-numeric elements requires an initial argument. (called from base_total at /Users/nikolayponomarev/projects/xxx/xxxx:25)
```
and following warning if I use `.sum(0)`
```
Argument 0 is redundant because it is implied by default.Style/RedundantArgument
```
What should we use `.sum` or `.sum(0)` ?
